csrc/optimization/concretize_symbolic_root.cpp

```diff
@@ -0,0 +1,75 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#include <debug.h>
+#include <ir/utils.h>
+#include <optimization/concretize_symbolic_root.h>
+#include <options.h>
+#include <root_domain_map.h>
+
+namespace nvfuser::optimization {
+
+namespace {
+
+void concretizeSymbolicRootDomain(Fusion* fusion) {
+  // map non-const extents to const extents
+  std::unordered_map<Val*, Val*> replacement_map;
+
+  const auto mapped_sets = ExactRootDomainMap(fusion).getMappedSets();
+  // Loop over each exact root domain set
+  for (const auto& set_ptr : mapped_sets.disjointSets()) {
+    // For each set, find an extent that is a const scalar
+    Val* const_extent = nullptr;
+    for (auto id : *set_ptr) {
+      // Skip broadcast.
+      // Skip non-root domain, otherwise, ConcretizeConstantExtents fails
+      // due to allocation domains included in ExactRootDomainMap.
+      if (id->isBroadcast() || id->definition()) {
+        continue;
+      }
+      if (id->extent()->isConstScalar()) {
+        const_extent = id->extent();
+        break;
+      }
+    }
+
+    // If no const extent was found, skip this set
```

Comment:
You might want to remove this check and instead pick the symbolic `Val` with the lowest `name()` as the representative here. This could simplify some cases where we recompute expressions inside the kernel that are known to be equal, even if they are not constant.

Reply:
added. The new logic is 
(1) if has const, replace with const 
(2) if no const, reaplace with `Val` with the lowest `name() `